workspaces/adventure-pack/goodies/python3/src/list_transpose/test.py

```diff
@@ -0,0 +1,43 @@
+from . import *
+
+
+def test_improper_input() -> None:
+    assert len(list_transpose([])) == 0
+    assert len(list_transpose([[], []])) == 0
+    assert len(list_transpose([[0], [0, 1]])) == 0
+
+
+def assert_matrices_equal(list1, list2):
+    assert len(list1) == len(list2)
+    for i in range(len(list1)):
+        assert len(list1[i]) == len(list2[i])
+        for j in range(len(list1[i])):
+            assert list1[i][j] == list2[i][j]
+
+
+def test_nonsymetric_transpose() -> None:
+    input_1 = [[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]]
```

Comment:
Despite my comment about less predictable test data in #370 I actually like this test case because I think it reads really well. But I suppose I wouldn't mind a test case with more shuffled data in addition to this one.